json/src/fs2/data/json/package.scala

```diff
@@ -51,8 +51,36 @@ package object json {
   def transform[F[_], Json](selector: Selector, f: Json => Json)(implicit F: RaiseThrowable[F],
                                                                  builder: Builder[Json],
                                                                  tokenizer: Tokenizer[Json]): Pipe[F, Token, Token] =
+    TokenSelector.transformPipe[F, Json](selector, f.andThen(Option(_)))
```

Comment:
```suggestion
    TokenSelector.transformPipe[F, Json](selector, f.andThen(Some(_)))
```
No need for checking against `null` values, I suppose?

Reply:
And why not? This is user code, and user could return whatever, so I think better safe than sorry.

Reply:
I leave being overly smart to frameworks and tend to KISS libraries. There's a clear way of removing values, so I don't see why someone should return `null` here. So why have everybody pay the (small) price?

Side note: Just looking at the code, `null` would actually bubble through to the final `Stream` without exploding, right? That could be used for interesting hacks like counting the amount of removed elements (of course no FP would do it) ;)

Seeing good arguments for both and having only a slight tendency myself, I leave the decision to you.

Reply:
I don't see the price as big, and as a functional library I would expect no `null` to be emitted. I prefer not allowing hacks based on `null` and extend and fix library if the needs exists. Also to make it clear, I will move the selector based pipes into the `selector` package and deprecate the ones in the `json` package for now (to be removed in 1.0.0). This will allow for other selector things based on another approach without having these selector always in scope. I can change the behavior only in the new package, then.